### ISSUE TITLE
Importing saved records with serialized fields

### ISSUE THREAD

**Issue by Andreis13** [zdennis/activerecord-import#424]
When importing ActiveRecord objects, there is a difference in behaviour when importing newly initialized objects versus objects that have been read from the database.

For instance, when a record initialized from arguments is imported, serialized attributes behave as expected: 
```ruby
new_record = Widget.new(json_data: { a: :b })
Widget.import([new_record])

Widget.last.json_data #=> { a: :b }
```

Meanwhile, if we read the object from the database and import it (this use-case came up when trying to copy data from one database to another), the JSON attribute is not persisted properly:
```ruby
existing_record = Widget.last
Widget.import([existing_record])

Widget.last.json_data #=> "{\"a\":\"b\"}"
```

The reason for such difference in behaviour is the way `read_attribute_before_type_cast` works for new, and persisted records respectively.

```ruby
Widget.new(json_data: { a: :b }).read_attribute_before_type_cast(:json_data) #=> { a: :b }
Widget.last.read_attribute_before_type_cast(:json_data) #=> "{\"a\":\"b\"}"
```

As far as I understand, the use of `read_attribute_before_type_cast` [in the code](https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/import.rb#L447), is primarily for performance reasons, so using only `read_attribute` wouldn't be a desirable option. Is it possible to somehow handle such a use-case in the future releases of the gem?

**Comment by daslicious** [comment]
`Widget.new(json_data: { a: :b })` will import with `json_data` being empty for me.

I'm working around it with 

```
widget = Widget.new(json_data: { a: :b })
widget.json_data = widget.json_data
widget
```

**Comment by Andreis13** [comment]
On our project such a workaround wouldn't work, because we have to import multiple arbitrary models in bulk. A simplified version of what we have would look like this:
```ruby
[SomeModel, SomeOtherModels, ThirdModels].each do |model|
  use_source_database_connection!
  records = model.all

  use_destination_database_connection!
  model.import(records)
end
```

In order to solve the issue with serialization we had to fork&patch. We just opted in for always using `read_attribute` instead of `read_attribute_before_type_cast`.

**Comment by jkowens** [comment]
@Andreis13 what if you specify your serialized attribute as a [stored attribute](http://api.rubyonrails.org/classes/ActiveRecord/Store.html)? ar-import knows to use `read_attribute` for those.

https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/import.rb#L445

**Comment by Andreis13** [comment]
@jkowens stored attributes have somewhat different behaviour even though it uses `serialize` under the hood. The attributes appear in the `stored_attributes` list only if you define accessors.
Consider the following example:
```ruby
class Bar < ApplicationRecord
  store :store_field, coder: JSON
end
#---------------------
pry(main)> Bar.stored_attributes #=> {}
```

In this situation, defining an arbitrary accessor for a stored attribute, so that it appears in the list, is just a hack that pollutes the model with unnecessary methods.

**Comment by jkowens** [comment]
@Andreis13 I've merged a fix into master. Can you give it a spin and let me know how it goes?

**Comment by Andreis13** [comment]
@jkowens yeah, this issue is solved now, thanks a lot!
However, I've identified another problem related to the behaviour of `read_attributes_before_type_cast` when used on timestamp fields, I'll open a separate issue.

**Comment by jkowens** [comment]
This fix has been released in v0.19.0.
